### PR TITLE
Actualize pricing module

### PR DESCRIPTION
**Pull request by AliveMen**
#45 Increase precision of Price money amounts up to 4 decimal digits

Pricing module UX improvements #40 (#41)
Price list assignments blade changes:
- Table width was increased;
- Added Activation and Expiration dates columns;
- Added search filter;
- Introduced ability to delete all filtered assignments;

add virtual GetAllPricelistAssignments for overriding (#35)

Removed the "Error: undefined" message when removing pricelists, made the blade to refresh its contents properly after removing. (#39)

### CODE REVIEW

File: Modules/vc-module-pricing/VirtoCommerce.PricingModule.Web/Module.cs [hunk 1]
```diff
@@ -88,24 +92,30 @@ public void PostInitialize(IApplicationBuilder appBuilder)
             var mvcJsonOptions = appBuilder.ApplicationServices.GetService<IOptions<MvcJsonOptions>>();
             mvcJsonOptions.Value.SerializerSettings.Converters.Add(appBuilder.ApplicationServices.GetService<PolymorphicPricingJsonConverter>());
 
-            // Add price document source to the product indexing configuration
-            var productIndexingConfigurations = appBuilder.ApplicationServices.GetServices<IndexDocumentConfiguration>();
-            if (productIndexingConfigurations != null)
+            var priceIndexingEnabled = settingsManager.GetValue("Pricing.Indexing.Enable", true);
```
Comment (tatarincev) [inline Modules/vc-module-pricing/VirtoCommerce.PricingModule.Web/Module.cs]: Use setting name from strongly type constant

File: Modules/vc-module-pricing/VirtoCommerce.PricingModule.Data/Search/ProductPriceDocumentChangesProvider.cs [hunk 2]
```diff
@@ -83,11 +91,55 @@ public virtual async Task<IList<IndexDocumentChange>> GetChangesAsync(DateTime?
...
+                    _settingsManager.GetValue("VirtoCommerce.Search.IndexingJobs.IndexationDate.Pricing.Calendar",
+                        (DateTime?)null) ?? DateTime.MinValue;
+                if (DateTime.UtcNow - lastIndexDate > CalendarChangesInterval && startDate != null && endDate != null)
+                {
+                    var calendarChanges =
+                        await GetCalendarChangesAsync(startDate.Value, endDate.Value, workSkip, workTake);
+
+                    if (workTake > 0)
+                    {
+                        _settingsManager.SetValue("VirtoCommerce.Search.IndexingJobs.IndexationDate.Pricing.Calendar", DateTime.UtcNow);
```
Comment (tatarincev) [inline Modules/vc-module-pricing/VirtoCommerce.PricingModule.Data/Search/ProductPriceDocumentChangesProvider.cs]: Use please strongly type settings from constants

File: Modules/vc-module-pricing/VirtoCommerce.PricingModule.Data/Search/ProductPriceDocumentChangesProvider.cs [hunk 3]
```diff
@@ -83,11 +91,55 @@ public virtual async Task<IList<IndexDocumentChange>> GetChangesAsync(DateTime?
...
 
+        #region IPricingDocumentChangesProvider Members
+        public virtual Task<IList<IndexDocumentChange>> GetCalendarChangesAsync(DateTime? startDate, DateTime? endDate, int skip = 0, int take = 0)
+        {
+            IList<IndexDocumentChange> result = null;
+
+            using (var repository = _repositoryFactory())
+            {
+                //Return calendar changes only for prices that have at least one specific date range.
+                var priceLists = repository.PricelistAssignments.Where(x => x.StartDate != null || x.EndDate != null)
```
Comment (tatarincev) [inline Modules/vc-module-pricing/VirtoCommerce.PricingModule.Data/Search/ProductPriceDocumentChangesProvider.cs]: Use to ArrayAsync() to execute query

File: Modules/vc-module-pricing/VirtoCommerce.PricingModule.Data/Search/ProductPriceDocumentChangesProvider.cs (hunk 2 above)
Comment (AliveMen) [reply Modules/vc-module-pricing/VirtoCommerce.PricingModule.Data/Search/ProductPriceDocumentChangesProvider.cs]: fixed

File: Modules/vc-module-pricing/VirtoCommerce.PricingModule.Data/Search/ProductPriceDocumentChangesProvider.cs (hunk 3 above)
Comment (AliveMen) [reply Modules/vc-module-pricing/VirtoCommerce.PricingModule.Data/Search/ProductPriceDocumentChangesProvider.cs]: fixed

File: Modules/vc-module-pricing/VirtoCommerce.PricingModule.Web/Module.cs (hunk 1 above)
Comment (AliveMen) [reply Modules/vc-module-pricing/VirtoCommerce.PricingModule.Web/Module.cs]: fixed

**Pull request merged**